kernel/src/hil/hw_debug.rs

```diff
@@ -0,0 +1,31 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2022.
+
+//! Interfaces for interacting with debug hardware integrated in various SoCs.
+//! Currently allows reading the cycle counter.
+
+pub trait CycleCounter {
+    /// Enable and start the cycle counter.
+    fn start(&self);
+
+    /// Stop the cycle counter.
+    /// Does nothing if the cycle counter is not present.
+    fn stop(&self);
+
+    /// Return the current value of the cycle counter.
+    fn count(&self) -> u32;
+
+    /// Reset the counter to zero and stop the cycle counter.
+    fn reset(&self);
```

Comment:
Does `reset()` need to be called before `start()`?

Reply:
Not necessarily! If counting cumulative cycles in a closure or something, one might want to stop and start the cycle counter without resetting it to avoid having to use a separate accumulator

Reply:
I meant the first time at startup

Reply:
Added a comment